encord/objects/ontology_labels_impl.py

```diff
@@ -1236,10 +1236,7 @@ def remove_classification(self, classification_instance: ClassificationInstance)
 
         classification_hash = classification_instance.classification_hash
         self._classifications_map.pop(classification_hash)
-        all_frames = self._classifications_to_frames[classification_instance.ontology_item]
-        actual_frames = _frame_views_to_frame_numbers(classification_instance.get_annotations())
-        for actual_frame in actual_frames:
-            all_frames.remove(actual_frame)
+        self._classifications_to_frames[classification_instance.ontology_item] = set()
```

Comment:
This doesn't look right to me. 

I'm not super familiar with the inner structure of the new label rows, so we may need to involve Denis here, but from the general principles it looks like a case of confusion between "instances" and "types". We're removing an *instance* of the classification, but the new line removed all the mapping for the whole *type*; this mapping is maintained, AFAICS, to guarantee uniqueness (you can't have two instances of the same classification on any given frame).

The assertion in the test is valid; I don't seem to have access to the original complaint but I suspect user error (which may indicate that our API is too complicated for the intended audience). Can you link or copy-paste the customer notebook itself to the PR or Jira ticket?

Reply:
Yep I don't think this fixes the underlying issue as shown by the tests. Going to dig into the tests and loop in Sergei to get to the bottom of it.

This is the link to the files sent by the customer:
https://drive.google.com/drive/folders/1zx0wo2JCFg4nnIOhbvN4Z9HRpW9uNUp_?usp=share_link